test/Serilog.Sinks.Raygun.Tests/Sinks/Raygun/RaygunClientSinkTests.cs

```diff
@@ -0,0 +1,150 @@
+        private const string OccurredProperty = "RaygunSink_OccurredOn";
+        private const string RaygunUserInfoPropertyName = "RaygunSink_UserInfo";
+
+        [SetUp]
+        public void Setup()
+        {
+            // _mockRaygunClient = new Mock<RaygunClientBase>(); // Removed
+            // _capturedMessage = null; // Removed
+
+            var messageTemplateParser = new MessageTemplateParser();
+            _logEvent = new LogEvent(
+                timestamp: DateTimeOffset.UtcNow,
+                level: LogEventLevel.Information,
+                exception: null,
+                messageTemplate: messageTemplateParser.Parse("Test message"),
+                properties: new List<LogEventProperty>());
+        }
+
+        // Helper to simulate property dictionary creation (simplified Emit)
+        private IDictionary<string, object> GetPropertiesDictionary(LogEvent logEvent, Func<LogEvent, RaygunIdentifierMessage?>? userInfoCallback)
+        {
+            var properties = logEvent.Properties.ToDictionary(kv => kv.Key, kv => kv.Value as object);
+
+            properties[RenderedLogMessageProperty] = logEvent.RenderMessage(null);
+            properties[LogMessageTemplateProperty] = logEvent.MessageTemplate.Text;
+            properties[OccurredProperty] = logEvent.Timestamp.UtcDateTime;
+
+            if (userInfoCallback != null)
+            {
+                var userInfo = userInfoCallback(logEvent);
+                if (userInfo != null)
+                {
+                    // Simulate how user info is stored before processing
+                    var structureProps = new List<LogEventProperty>();
+                    if (userInfo.Identifier != null) structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.Identifier), new ScalarValue(userInfo.Identifier)));
+                    structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.IsAnonymous), new ScalarValue(userInfo.IsAnonymous)));
+                    if (userInfo.Email != null) structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.Email), new ScalarValue(userInfo.Email)));
+                    if (userInfo.FullName != null) structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.FullName), new ScalarValue(userInfo.FullName)));
+                    if (userInfo.FirstName != null) structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.FirstName), new ScalarValue(userInfo.FirstName)));
+                    if (userInfo.UUID != null) structureProps.Add(new LogEventProperty(nameof(RaygunIdentifierMessage.UUID), new ScalarValue(userInfo.UUID)));
+                    properties[RaygunUserInfoPropertyName] = new StructureValue(structureProps, nameof(RaygunIdentifierMessage));
+                }
+            }
+            return properties;
+        }
+
+        [Test]
+        public void UserInfoCallback_ProvidedAndReturnsUser_SetsUserDetails()
```

Comment:
The first part of these test names should be "ProcessRaygunMessageDetails" - the actual method being acted on under test.